scripts/ci/make/pipeline_tasks

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env bash
+
+# source utility functions
+# shellcheck source=SCRIPTDIR/utils
+source "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/utils"
+
+##########################################################
+###   PROJECT SPECIFIC TASK COMMANDS - RUNTIME ORDER   ###
+########################################################
+
+# These functions are intended to be utilized as task commands invoked from the run_make_task script
+
+build() {
+    local COMMIT_SHA="${1:?'required parameter'}"
+    local TEST_IMAGE_NAME="${2:?'required parameter'}"
+
+    print_colorized WARN "building image ${TEST_IMAGE_NAME}"
+    docker build --target anchore-cli-builder -t anchore-cli:builder -f ./Dockerfile .
+    docker build --build-arg CLI_COMMIT="${COMMIT_SHA}" -t "${TEST_IMAGE_NAME}" -f ./Dockerfile .
+    print_colorized WARN "successfully built image -- ${TEST_IMAGE_NAME}"
+}
+
+lint() {
+    local PYTHON="${1:?'required parameter'}"
+    local VENV="${2:?'required parameter'}"
+
+    activate_venv "${VENV}"
+    hash tox || "${PYTHON}" -m pip install -q tox
+
+    print_colorized WARN "linting project"
+    tox -e flake8
+}
+
+unit_tests() {
+    local PYTHON="${1:?'required parameter'}"
+    local VENV="${2:?'required parameter'}"
+
+    # Declare default tox environments, but allow TOX_ENV environment variable to override if it's set
+    local default_tox_envs="py27 py36"
```

Comment:
Ok, so you are defining default environments in the script. I generally don't mind having helpers but I would prefer for these configurations to be as close to the tooling as possible. 

To me, the process looks like this:

CircleCi reads config -> executes a make command -> a script function is called -> tox is executed

For a Python dev, having a script function define what needs to run is going to be surprising. This is a CI configuration, and I am not clear why these aren't defined there.

Reply:
The defaults in the script are for running the pipeline locally. In CI it uses parameters in the workflow job declarations.

Reply:
Oh I see the disconnect here. I removed the tox_env parameters from the job because I wanted unit tests to run in a single job. I will add them back to make it more clear.

Reply:
> Oh I see the disconnect here. I removed the tox_env parameters from the job because I wanted unit tests to run in a single job. I will add them back to make it more clear.

Yeah, that was it. Thanks for taking care of adding them back